src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2860,6 +2850,26 @@ private BoundExpression BindArrayCreationExpression(ArrayCreationExpressionSynta
                 : BindArrayCreationWithInitializer(diagnostics, node, node.Initializer, type, arraySizes);
         }
 
+        private BoundExpression BindArrayDimension(ExpressionSyntax dimension, SyntaxNode node, DiagnosticBag diagnostics)
+        {
+            // These make the parse tree nicer, but they shouldn't actually appear in the bound tree.
+            if (dimension.Kind() != SyntaxKind.OmittedArraySizeExpression)
+            {
+                var size = BindValue(dimension, diagnostics, BindValueKind.RValue);
+                if (!size.HasAnyErrors)
+                {
+                    size = ConvertToArrayIndex(size, node, diagnostics, allowIndexAndRange: false);
```

Comment:
>node [](start = 53, length = 4)

Should this be ```dimension```? We probably don't need the ```node``` parameter at all

Reply:
As noted above, this is the current behaviour on master, but I will change it.